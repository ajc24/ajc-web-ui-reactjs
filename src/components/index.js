/**
 * Developed by Anthony Cox in 2024
 */

/* Export the header components */
export { default as HeaderTypeA } from './header/HeaderTypeA';

/* Export the main components */
export { default as MainContent } from './main/MainContent';

/* Export the menu bar components */
export { default as MenuBarTypeA } from './menu-bar/MenuBarTypeA';
