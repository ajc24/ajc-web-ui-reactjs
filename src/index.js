/**
 * Developed by Anthony Cox in 2024
 */

/* Import the header components */
import Header from './components/header/Header';

/* Import the image components */
import DecorativeImage from './components/images/DecorativeImage';
import FunctionalImage from './components/images/FunctionalImage';
import InformativeImage from './components/images/InformativeImage';

/* Export all components */
export {
  DecorativeImage,
  FunctionalImage,
  InformativeImage,
  Header,
};
