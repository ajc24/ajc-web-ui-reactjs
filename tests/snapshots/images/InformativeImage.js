/**
 * Developed by Anthony Cox in 2024
 */
import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-testing-code';
import { InformativeImage } from '../../../src';
import benjiLogo from '../../../src/components/images/files/logo-benji.png';

describe('Informative Image', () => {
  describe('With default parameters', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <InformativeImage id="informative-image-1" src={benjiLogo} alt="Default informative component alt text" />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });

  describe('With custom parameters', () => {
    let snapshot;
    
    beforeAll(() => {
      snapshot = TestDev.createSnapshot(
        <React.Fragment>
          <InformativeImage id="informative-image-2" src={benjiLogo} alt="Custom informative component alt text" height={300} width={500} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(snapshot).toMatchSnapshot();
    });
  });
});
