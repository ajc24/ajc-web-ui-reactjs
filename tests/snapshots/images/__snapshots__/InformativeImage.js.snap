// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Informative Image With custom parameters verifies the snapshot for the component 1`] = `"<img id="informative-image-2--informative-image" class="image" src="[object Object]" width="500" height="300" alt="Custom informative component alt text">"`;

exports[`Informative Image With default parameters verifies the snapshot for the component 1`] = `"<img id="informative-image-1--informative-image" class="image" src="[object Object]" width="100" height="100" alt="Default informative component alt text">"`;
